includes/admin/settings/register-settings.php

```diff
@@ -29,6 +29,35 @@ function edd_get_option( $key = '', $default = false ) {
 }
 
 /**
+ * Update an option
+ *
+ * Updates an edd setting value in both the db and the global variable.
+ *
+ * @since 2.3
+ * @return boolean True if updated, false if not.
+ */
+function edd_update_option( $key = '', $value = false ) {
+	// First let's grab the current settings
+	$options = get_option( 'edd_settings' );
+
+	// Let's let devs alter that value coming in
+	$value = apply_filters( 'edd_update_option', $value, $key );
+
+	// Next let's try to update the value
+	$options[ $key ] = $value;
```

Comment:
That sounds really dangerous. If you set a value to false, which alot of people do, including core, doing that means the index keys won't exist.
Before edd_get_option, extension developers used to use the edd_option global. Let's say we have a key called 'something'.

My code from before edd_get_option could have reasonably been:

if ( $edd_options['something] ){

}
because even if the value was set to false, I still expected the array key to be there.
Suddenly if we remove it we're causing errors. Sure, is it their responsibility to do an isset check? Maybe, but we can't assume people have